web/styles/left_sidebar.css

```diff
@@ -224,14 +219,115 @@ li.show-more-topics {
+    margin-left: 6px;
+    /* Space between #views-label-container and #left-sidebar-navigation-list in collapsed state. */
+    margin-right: 10px;
+    margin-top: 3px;
+    outline: none;
+}
 
-        &:hover {
+#left-sidebar-navigation-area {
+    & li.top_left_row {
+        position: relative;
+        padding: 2px 10px 1px $far_left_gutter_size;
+
+        & a:hover {
             text-decoration: none;
         }
     }
+
+    &.collapsed {
+        display: flex;
+        align-items: center;
+        /* 2px space between views and direct messages sections when views section is collapsed. */
+        margin-bottom: 2px;
+
+        /* Only show first 5 views in collapsed state. */
+        .top_left_row:nth-child(n + 6) {
+            display: none !important;
+        }
+
+        & #left-sidebar-navigation-list {
+            width: 100%;
+            display: flex;
+            /* 25px margin-right of .subscription_block + 12px margin-right of #stream_filters. */
+            margin-right: 37px;
+            margin-bottom: 0;
+            padding: 0;
+
+            & li {
+                padding: 0;
+                flex-grow: 1;
+
+                .left-sidebar-navigation-label-container {
+                    padding: 0;
+                    justify-content: center;
+
+                    .filter-icon {
+                        padding-top: 4px;
+
+                        & i {
+                            padding-right: 0;
+
+                            &.fa {
+                                font-size: 16px;
+                            }
+
+                            &.zulip-icon-star-filled {
+                                top: 1px;
+                            }
+                        }
+                    }
+
+                    .left-sidebar-navigation-label {
+                        display: none;
+                    }
+                }
+
+                .arrow {
+                    display: none;
+                }
+            }
+        }
+
+        /* CSS to align unread counters in top right corner. */
+        .unread_count {
+            position: absolute;
+            bottom: calc(100% - 4px);
+            left: calc(100% - 4px);
+            z-index: 3;
+            transform: translate(-25%, 25%);
+        }
+
+        /* Until hovered on li, show unread counters as dots. */
+        & li:not(:hover) .unread_count {
+            width: 8px;
+            height: 8px;
+            border: 0;
+            padding: 0;
+            font-size: 0;
+            transform: translate(0%, 0%);
+            background-color: var(--color-background-unread-counter);
+        }
+
+        & .left-sidebar-navigation-menu-icon {
+            display: block;
+            cursor: pointer;
+            top: auto;
+            /* 12px margin-right of #stream_filters to align with other vertical 3-dots in left sidebar. */
+            margin-right: 12px;
+            color: var(--color-global-filter-icon);
```

Comment:
Since you've (thankfully) gotten rid of the `global_filters` references in this PR, we should rename it (I think this is also referenced around line 62 above). 

For the vdots here, I'm thinking you ought to specify something like `--color-vdots-navigation-menu`, and just assign it to the renamed global-filter icon style to keep those in sync.

Hopefully in the redesign, that additional vdots variable becomes unnecessary and we can use one of the existing vdots colors so we don't go back to having so many different colors for them.